renderdoc/driver/vulkan/vk_acceleration_structure.h

```diff
@@ -37,7 +43,18 @@ class VulkanAccelerationStructureManager
     bool isTLAS;
   };
 
-  VulkanAccelerationStructureManager(WrappedVulkan *driver) : m_pDriver(driver) {}
+  explicit VulkanAccelerationStructureManager(WrappedVulkan *driver);
+
+  // ResourceId.  BDA isn't known at creation time and the usage flags aren't known at bind time so
+  // initialise is a two-step process.  These should only be called when capturing
+  void TrackInputBuffer(VkDevice device, VkResourceRecord *record);
+  void UntrackInputBuffer(VkResourceRecord *record);
```

Comment:
Rather than having a new implementation of this, I think it would be better to move D3D12's address tracking to a new common location and then use it here.

If you look at `GPUAddressRangeTracker` in `d3d12_manager.h` it should be fairly easy to lift out - `D3D12_GPU_VIRTUAL_ADDRESS` is a typedef for `uint64_t` the same as `VkDeviceAddress` so we can define an API-agnostic typedef and use it with implicit conversions. We don't have much that is shared between APIs but it can be moved to a new pair of files next to `intervals.h/cpp`.

Also the likelihood is high that in the future we'll need to do this kind of lookup if the API design continues its trend - can you put this tracking inside `WrappedVulkan` itself rather than tying it to acceleration structures as this in itself is not AS-specific.